src/components/Pagination.vue

```diff
@@ -185,7 +185,12 @@ export default {
         },
 
         rangeChange () {
-            this.$emit('range-change', this.start, this.end);
+            if (this.page >= this.totalPages && this.totalPages !== 0) {
+                this.$emit('page-change', this.totalPages - 1);
+            }
+            else {
+                this.$emit('range-change', this.start, this.end);
```

Comment:
This change assumes that the parent component will update `page` property in response to `'page-change'` event (that's how the page buttons work). As a result `pageChange` will be called that calls `rangeChange` that now falls into the `else` branch.